subgraphs/_reference_/src/sdk/protocols/bridge/index.ts

```diff
@@ -85,6 +31,46 @@ export class SDK {
     this.Protocol.sdk = this;
   }
 
+  static initializeFromEvent(
+    config: BridgeConfigurer,
+    pricer: TokenPricer,
+    tokenInitializer: TokenInitializer,
+    event: ethereum.Event
+  ): SDK {
+    const customEvent = CustomEventType.initialize(
+      event.block,
+      event.transaction,
+      event.logIndex,
+      event
+    );
+    return new SDK(config, pricer, tokenInitializer, customEvent);
+  }
+
+  static initializeFromCall(
+    config: BridgeConfigurer,
+    pricer: TokenPricer,
+    tokenInitializer: TokenInitializer,
+    event: ethereum.Call
+  ): SDK {
+    const customEvent = CustomEventType.initialize(
+      event.block,
+      event.transaction,
+      BIGINT_ZERO
+    );
+    return new SDK(config, pricer, tokenInitializer, customEvent);
+  }
+
+  static intializeFromCustomEvent(
+    config: BridgeConfigurer,
+    pricer: TokenPricer,
+    tokenInitializer: TokenInitializer
+  ): SDK {
+    return new SDK(config, pricer, tokenInitializer, new CustomEventType());
```

Comment:
Ah my mistake, I believe I put in the proper fix.

Reply:
Same thing on `generic/index.ts` :D

Reply:
<img width="556" alt="Screen Shot 2023-01-26 at 1 57 02 AM" src="https://user-images.githubusercontent.com/56660047/214784389-6f278d6e-a198-439e-b3ae-492f076bc056.png">

Oh wait, I did mean to put that there. I was following your lead with this in the Bridge schema. I guess its for chains that do not have a normal ethereum.Event/Call data type?

Reply:
Interesting. In this case we can panic, do some `log.critical` if that type is not considered, since it is a misconfiguration and not something that will come up in the middle of the sync process